synapse/handlers/room_member.py

```diff
@@ -793,12 +773,38 @@ def _lookup_3pid_v2(
             id_access_token (str): The access token to authenticate to the identity server with
             medium (str): The type of the third party identifier (e.g. "email").
             address (str): The third party identifier (e.g. "foo@example.com").
-            hash_details (dict[str, str|list]): A dictionary containing hashing information
-                provided by an identity server.
 
         Returns:
             Deferred[str|None]: the matrix ID of the 3pid, or None if it is not recognised.
         """
+        try:
+            # Check what hashing details are supported by this identity server
+            hash_details = yield self.simple_http_client.get_json(
+                "%s%s/_matrix/identity/v2/hash_details" % (id_server_scheme, id_server),
+                {"access_token": id_access_token},
+            )
+        except HttpResponseException as e:
+            if e.code == 404:
+                raise
+
+            logger.warning("Error when performing a v2 hash_details request: %s", e)
+            raise SynapseError(
```

Comment:
why do we not just allow the original exception to propagate? it looks like it will be correctly handled?